### PR TITLE
purple-slack: Init at 2019-08-18

### PR DESCRIPTION
**Pull request by eyJhb**
###### Motivation for this change
Wasn't in the packages.

###### Things done
Tested using Bitlbee, works fine

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @tilpner you did most of the work, maybe you can review and make sure everything is OK.
Anyone else is welcome as well !

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  version = "unstable-2019-08-18";
+
+  src = fetchFromGitHub {
+    owner = "dylex";
+    repo = "slack-libpurple";
+    rev = "be97802c7fd0b611722d2f551756e2a2672f6084";
+    sha256 = "0l6hwnnv6zzszxkk0a3nli88w5gijvlc9qgkkai0sb4v4c504y5v";
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
```
Comment (tilpner) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]:
```suggestion
```

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  buildInputs = [ pidgin ];
+
+  PKG_CONFIG_PURPLE_PLUGINDIR = "${placeholder "out"}/lib/purple-2";
+  PKG_CONFIG_PURPLE_DATAROOTDIR = "${placeholder "out"}/share";
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/dylex/slack-libpurple;
+    description = "Slack plugin for Pidgin";
+    license = licenses.gpl2;
+    platforms = platforms.linux;
```
Comment (tilpner) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]: Are you sure?

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitHub, pkgconfig, pidgin, json-glib }:
```
Comment (tilpner) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]:
```suggestion
{ stdenv, fetchFromGitHub, pidgin }:
```

**Review by tilpner — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix (hunk 2 above)
Comment (eyJhb) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]: https://github.com/dylex/slack-libpurple/blob/master/COPYING ? States GPLv2

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix (hunk 3 above)
Comment (eyJhb) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]: Resolved in latest commit.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix (hunk 1 above)
Comment (eyJhb) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]: Resolved in latest commit.

**Comment by eyJhb** [comment]
@GrahamcOfBorg build purple-slack

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub, pidgin }:
+
+stdenv.mkDerivation {
+  pname = "purple-slack";
+  version = "unstable-2019-08-18";
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]:
```suggestion
  version = "2019-08-18";
```

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub, pidgin }:
+
+stdenv.mkDerivation {
+  pname = "purple-slack";
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]:
```suggestion
  pname = "purple-slack-unstable";
```

Comment (worldofpeace) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]: See: https://github.com/NixOS/nixpkgs/issues/68531

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  version = "unstable-2019-08-18";
+
+  src = fetchFromGitHub {
+    owner = "dylex";
+    repo = "slack-libpurple";
+    rev = "be97802c7fd0b611722d2f551756e2a2672f6084";
+    sha256 = "0l6hwnnv6zzszxkk0a3nli88w5gijvlc9qgkkai0sb4v4c504y5v";
+  };
+
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]:
```suggestion
  nativeBulidInputs = [ 
    pkgconfig
  ];
```

^ above too

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix (hunk 4 above)
Comment (eyJhb) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]: Should be resolved in latest force push :)

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix (hunk 5 above)
Comment (eyJhb) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]: Should be resolved in latest force push :)

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix (hunk 6 above)
Comment (eyJhb) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-slack/default.nix]: Should be resolved in latest force push :)

**Review by worldofpeace — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**